stress: failed test in cockroach/kv/kv.test: TestTxnDelRangeIntentResolutionCounts

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/dace3cde37c82d43f2f2c73ddf8c82a454b1e0b3

Stress build found a failed test:

```
=== RUN   TestTxnDelRangeIntentResolutionCounts
I161010 08:19:32.602432 15110 storage/engine/rocksdb.go:349  opening in memory rocksdb instance
I161010 08:19:32.604582 15110 gossip/gossip.go:234  [n?] initial resolvers: []
W161010 08:19:32.604599 15110 gossip/gossip.go:1045  [n?] no resolvers found; use --join to specify a connected node
W161010 08:19:32.605577 15110 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I161010 08:19:32.606064 15110 server/node.go:410  [n?] store [n0,s0] not bootstrapped
I161010 08:19:32.607476 15228 storage/replica_trigger.go:252  [s1,r1:/M{in-ax},r1:/M{in-ax}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 410024h19m41.606828803s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=2016-10-10 08:19:32.607440574 +0000 UTC]
I161010 08:19:32.607976 15110 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161010 08:19:32.608055 15110 server/node.go:342  [n?] **** cluster {e6a6446c-b3bb-4240-899c-39af9a3ce5bb} has been created
I161010 08:19:32.608066 15110 server/node.go:343  [n?] **** add additional nodes by specifying --join=127.0.0.1:33897
I161010 08:19:32.608322 15110 gossip/gossip.go:268  [n?] NodeID set to 1
I161010 08:19:32.608776 15110 storage/store.go:1042  [n?] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161010 08:19:32.608791 15110 server/node.go:423  [n?] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:0}
I161010 08:19:32.608814 15110 server/node.go:315  [n?] node ID 1 initialized
I161010 08:19:32.608848 15110 gossip/gossip.go:282  [n?] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33897" > attrs:<> 
I161010 08:19:32.608899 15110 storage/stores.go:292  [n?] read 0 node addresses from persistent storage
I161010 08:19:32.608951 15110 server/node.go:553  [n?] connecting to gossip network to verify cluster ID...
I161010 08:19:32.609191 15110 server/node.go:574  [n?] node connected via gossip and verified as part of cluster {"e6a6446c-b3bb-4240-899c-39af9a3ce5bb"}
I161010 08:19:32.609218 15110 server/node.go:361  [n?] node=1: started with [[]=] engine(s) and attributes []
I161010 08:19:32.609238 15110 server/server.go:526  [n1] starting https server at 127.0.0.1:34670
I161010 08:19:32.609250 15110 server/server.go:527  [n1] starting grpc/postgres server at 127.0.0.1:33897
I161010 08:19:32.609257 15110 server/server.go:528  [n1] advertising CockroachDB node at 127.0.0.1:33897
I161010 08:19:32.609853 14960 sql/event_log.go:94  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:33897} Attrs:} ClusterID:{UUID:e6a6446c-b3bb-4240-899c-39af9a3ce5bb} StartedAt:1476087572609204879}
I161010 08:19:32.612355 15330 storage/split_queue.go:103  [split,n1,s1,r1:/M{in-ax}] splitting at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I161010 08:19:32.613269 14960 sql/event_log.go:94  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:33897} Attrs:} ClusterID:{UUID:e6a6446c-b3bb-4240-899c-39af9a3ce5bb} StartedAt:1476087572609204879}
I161010 08:19:32.613708 15330 storage/replica_command.go:2328  [split,n1,s1,r1:/M{in-ax},r1:/M{in-ax}] initiating a split of this range at key /Table/11 [r2]
I161010 08:19:32.667131 15330 storage/replica_command.go:2328  [split,n1,s1,r1:/{Min-Table/11},r2:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r3]
I161010 08:19:32.673016 15330 storage/replica_command.go:2328  [split,n1,s1,r1:/{Min-Table/11},r3:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r4]
I161010 08:19:32.681331 15330 storage/replica_command.go:2328  [split,n1,s1,r1:/{Min-Table/11},r4:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r5]
I161010 08:19:32.726158 15110 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
--- FAIL: TestTxnDelRangeIntentResolutionCounts (0.13s)
	db_test.go:356: abortTxn: false, 1 intents left over
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
0 runs so far, 0 failures, over 25s
0 runs so far, 0 failures, over 30s
16 runs so far, 0 failures, over 35s
16 runs so far, 0 failures, over 40s
16 runs so far, 0 failures, over 45s
16 runs so far, 0 failures, over 50s
16 runs so far, 0 failures, over 55s
16 runs so far, 0 failures, over 1m0s
18 runs so far, 0 failures, over 1m5s
32 runs so far, 0 failures, over 1m10s
32 runs so far, 0 failures, over 1m15s
32 runs so far, 0 failures, over 1m20s
32 runs so far, 0 failures, over 1m25s
32 runs so far, 0 failures, over 1m30s
32 runs so far, 0 failures, over 1m35s
48 runs so far, 0 failures, over 1m40s
48 runs so far, 0 failures, over 1m45s
48 runs so far, 0 failures, over 1m50s
48 runs so far, 0 failures, over 1m55s
48 runs so far, 0 failures, over 2m0s
48 runs so far, 0 failures, over 2m5s
51 runs so far, 0 failures, over 2m10s
64 runs so far, 0 failures, over 2m15s
64 runs so far, 0 failures, over 2m20s
64 runs so far, 0 failures, over 2m25s
64 runs so far, 0 failures, over 2m30s
64 runs so far, 0 failures, over 2m35s
65 runs so far, 0 failures, over 2m40s
78 runs so far, 0 failures, over 2m45s
80 runs so far, 0 failures, over 2m50s
80 runs so far, 0 failures, over 2m55s
80 runs so far, 0 failures, over 3m0s
80 runs so far, 0 failures, over 3m5s
81 runs so far, 0 failures, over 3m10s
83 runs so far, 0 failures, over 3m15s
96 runs so far, 0 failures, over 3m20s
96 runs so far, 0 failures, over 3m25s
96 runs so far, 0 failures, over 3m30s
96 runs so far, 0 failures, over 3m35s
96 runs so far, 0 failures, over 3m40s
97 runs so far, 0 failures, over 3m45s
108 runs so far, 0 failures, over 3m50s
112 runs so far, 0 failures, over 3m55s
112 runs so far, 0 failures, over 4m0s
112 runs so far, 0 failures, over 4m5s
112 runs so far, 0 failures, over 4m10s
113 runs so far, 0 failures, over 4m15s
113 runs so far, 0 failures, over 4m20s
128 runs so far, 0 failures, over 4m25s
128 runs so far, 0 failures, over 4m30s
128 runs so far, 0 failures, over 4m35s
128 runs so far, 0 failures, over 4m40s
128 runs so far, 0 failures, over 4m45s
129 runs so far, 0 failures, over 4m50s
139 runs so far, 0 failures, over 4m55s
144 runs so far, 0 failures, over 5m0s
144 runs so far, 0 failures, over 5m5s
144 runs so far, 0 failures, over 5m10s
144 runs so far, 0 failures, over 5m15s
145 runs so far, 0 failures, over 5m20s
146 runs so far, 0 failures, over 5m25s
160 runs so far, 0 failures, over 5m30s
160 runs so far, 0 failures, over 5m35s
160 runs so far, 0 failures, over 5m40s
160 runs so far, 0 failures, over 5m45s
161 runs so far, 0 failures, over 5m50s
161 runs so far, 0 failures, over 5m55s
171 runs so far, 0 failures, over 6m0s
176 runs so far, 0 failures, over 6m5s
176 runs so far, 0 failures, over 6m10s
176 runs so far, 0 failures, over 6m15s
177 runs so far, 0 failures, over 6m20s
177 runs so far, 0 failures, over 6m25s
179 runs so far, 0 failures, over 6m30s
192 runs so far, 0 failures, over 6m35s
192 runs so far, 0 failures, over 6m40s
192 runs so far, 0 failures, over 6m45s
192 runs so far, 0 failures, over 6m50s
193 runs so far, 0 failures, over 6m55s
193 runs so far, 0 failures, over 7m0s
201 runs so far, 0 failures, over 7m5s
208 runs so far, 0 failures, over 7m10s
208 runs so far, 0 failures, over 7m15s
208 runs so far, 0 failures, over 7m20s
209 runs so far, 0 failures, over 7m25s
209 runs so far, 0 failures, over 7m30s
211 runs so far, 0 failures, over 7m35s
224 runs so far, 0 failures, over 7m40s
224 runs so far, 0 failures, over 7m45s
224 runs so far, 0 failures, over 7m50s
224 runs so far, 0 failures, over 7m55s
225 runs so far, 0 failures, over 8m0s
225 runs so far, 0 failures, over 8m5s
233 runs so far, 0 failures, over 8m10s
240 runs so far, 0 failures, over 8m15s
240 runs so far, 0 failures, over 8m20s
240 runs so far, 0 failures, over 8m25s
241 runs so far, 0 failures, over 8m30s
241 runs so far, 0 failures, over 8m35s
244 runs so far, 0 failures, over 8m40s
256 runs so far, 0 failures, over 8m45s
256 runs so far, 0 failures, over 8m50s
256 runs so far, 0 failures, over 8m55s
256 runs so far, 0 failures, over 9m0s
257 runs so far, 0 failures, over 9m5s
257 runs so far, 0 failures, over 9m10s
264 runs so far, 0 failures, over 9m15s
272 runs so far, 0 failures, over 9m20s
272 runs so far, 0 failures, over 9m25s
272 runs so far, 0 failures, over 9m30s
273 runs so far, 0 failures, over 9m35s
273 runs so far, 0 failures, over 9m40s
277 runs so far, 0 failures, over 9m45s
288 runs so far, 0 failures, over 9m50s
288 runs so far, 0 failures, over 9m55s
288 runs so far, 0 failures, over 10m0s
289 runs so far, 0 failures, over 10m5s
289 runs so far, 0 failures, over 10m10s
291 runs so far, 0 failures, over 10m15s
297 runs so far, 0 failures, over 10m20s
304 runs so far, 0 failures, over 10m25s
304 runs so far, 0 failures, over 10m30s
304 runs so far, 0 failures, over 10m35s
305 runs so far, 0 failures, over 10m40s
305 runs so far, 0 failures, over 10m45s
308 runs so far, 0 failures, over 10m50s
320 runs so far, 0 failures, over 10m55s
320 runs so far, 0 failures, over 11m0s
320 runs so far, 0 failures, over 11m5s
320 runs so far, 0 failures, over 11m10s
321 runs so far, 0 failures, over 11m15s
323 runs so far, 0 failures, over 11m20s
331 runs so far, 0 failures, over 11m25s
336 runs so far, 0 failures, over 11m30s
336 runs so far, 0 failures, over 11m35s
336 runs so far, 0 failures, over 11m40s
337 runs so far, 0 failures, over 11m45s
339 runs so far, 0 failures, over 11m50s
342 runs so far, 0 failures, over 11m55s
352 runs so far, 0 failures, over 12m0s
352 runs so far, 0 failures, over 12m5s
352 runs so far, 0 failures, over 12m10s
353 runs so far, 0 failures, over 12m15s
353 runs so far, 0 failures, over 12m20s
356 runs so far, 0 failures, over 12m25s
362 runs so far, 0 failures, over 12m30s
368 runs so far, 0 failures, over 12m35s
368 runs so far, 0 failures, over 12m40s
368 runs so far, 0 failures, over 12m45s
369 runs so far, 0 failures, over 12m50s
371 runs so far, 0 failures, over 12m55s
374 runs so far, 0 failures, over 13m0s
384 runs so far, 0 failures, over 13m5s
384 runs so far, 0 failures, over 13m10s
384 runs so far, 0 failures, over 13m15s
385 runs so far, 0 failures, over 13m20s
385 runs so far, 0 failures, over 13m25s
389 runs so far, 0 failures, over 13m30s
396 runs so far, 0 failures, over 13m35s
398 runs completed, 1 failures, over 13m35s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/pkg/kv/db_test.go
+++ b/pkg/kv/db_test.go
@@ -288,15 +288,19 @@ func TestAuthentication(t *testing.T) {
 func TestTxnDelRangeIntentResolutionCounts(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	var intentResolutionCount int64
-	var resolveIntentCount int64
 	params := base.TestServerArgs{
 		Knobs: base.TestingKnobs{
 			Store: &storage.StoreTestingKnobs{
 				NumKeysEvaluatedForRangeIntentResolution: &intentResolutionCount,
 				TestingCommandFilter: func(filterArgs storagebase.FilterArgs) *roachpb.Error {
-					_, ok := filterArgs.Req.(*roachpb.ResolveIntentRequest)
+					req, ok := filterArgs.Req.(*roachpb.ResolveIntentRequest)
 					if ok {
-						atomic.AddInt64(&resolveIntentCount, 1)
+						key := req.Header().Key.String()
+						// Check if the intent is from the range being
+						// scanned below.
+						if key >= "a" && key < "d" {
+							t.Errorf("resolving intent on key %s", key)
+						}
 					}
 					return nil
 				},
@@ -346,14 +350,11 @@ func TestTxnDelRangeIntentResolutionCounts(t *testing.T) {
 		}
 
 		// Ensure no intents are left behind. Scan the entire span to resolve
-		// any intents that were left behind.
-		atomic.StoreInt64(&resolveIntentCount, 0)
+		// any intents that were left behind; intents are resolved internally
+		// through ResolveIntentRequest(s).
 		kvs, err := db.Scan(context.TODO(), "a", "d", totalNumKeys)
 		if err != nil {
 			t.Fatal(err)
-		}
-		if c := atomic.LoadInt64(&resolveIntentCount); c != 0 {
-			t.Errorf("abortTxn: %v, %d intents left over", abortTxn, c)
 		}
 		expectedNumKeys := totalNumKeys / 2
 		if abortTxn {

